cloudferrylib/os/actions/vm_snapshot.py

```diff
@@ -35,10 +35,36 @@ def namespace_variable(self):
         """returns unique for cloud variable from namespace"""
         return '_'.join([VM_STATUSES, self.cloud.position])
 
-    def get_list_of_vms(self):
-        search_opts = {'all_tenants': 'True'}
-        return self.get_compute_resource().get_instances_list(
-            search_opts=search_opts)
+    def get_similar_tenants(self):
+        src_identity = self.src_cloud.resources[utils.IDENTITY_RESOURCE]
+        dst_identity = self.dst_cloud.resources[utils.IDENTITY_RESOURCE]
+        src_tenants = src_identity.read_info()['tenants']
+        dst_tenants = {t['tenant']['name']: t['tenant']['id']
+                       for t in dst_identity.read_info()['tenants']}
+        similar_tenants = {}
+        for ts in src_tenants:
+            index = ts['tenant']['name']
+            if index in dst_tenants:
+                similar_tenants[ts['tenant']['id']] = dst_tenants[index]
+            else:
+                similar_tenants[ts['tenant']['id']] = ''
+        return similar_tenants
+
+    def get_list_of_vms(self, search_opts_tenants):
+        is_dst = self.cloud.position == 'dst'
+        tenants = search_opts_tenants.get('tenant_id', None)
+        search_opts = search_opts_tenants.get('search_opts', {}) \
+            if not is_dst else {}
+        search_opts = search_opts if search_opts else {}
+        search_opts.update(all_tenants=True)
+        tenants_map = {}
+        if is_dst and tenants:
+            tenants_map = self.get_similar_tenants()
+        if tenants:
+            tenant_id = tenants_map[tenants[0]] if is_dst else tenants[0]
```

Comment:
Looks like the `self.get_similar_tenants()` function is redundant, you can just call `dst_identity.get_tenant_id_by_name()` here and update the `search_map` if tenant exists